src/app/common.service.ts

```diff
@@ -6,6 +6,40 @@ import { SharedService } from './shared.service';
   providedIn: 'root'
 })
 export class CommonService {
-
   constructor(private httpService: AppService, public sharedService:SharedService) { }
+
+  checkValidEmail(email) {
+       if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(email)) {
+           return true;
+       } else {
+         return false;
+       }
+   }
+
+  required(data){
+  	if (data == undefined || data == null || data == "") {
+  		return false;
+  	}else{
+  		return true;
+  	}
+  }
+
+  checkValidMobileNumber(mobileNumber) {
+       if(mobileNumber){
+
+	       if(mobileNumber.toString().length == 10){
+	        	return true;
+	       }else{ 
+	       		return false;
+	   		}
+       }
+	}
+
+  matchPassword(user){
```

Comment:
This function should be only in signup component. Those functions which are called from multiple components are to placed here and this will be only called one time from signup.